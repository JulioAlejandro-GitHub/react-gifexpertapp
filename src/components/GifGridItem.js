import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url, id}) => {
    return (
        <div className="card animate__animated animate__tada">
            <img
                src={url} 
                alt={title}
            />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    id: PropTypes.number
}
GifGridItem.defaultProps = {
    url : "https://",
    title : "Default Title",
    id : 1
}