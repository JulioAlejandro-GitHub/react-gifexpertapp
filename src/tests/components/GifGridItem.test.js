import React from "react";
import { shallow } from "enzyme";
import {GifGridItem} from "../../components/GifGridItem";

describe('Test <GifGridItem />', () => {

    const url = 'httls://sdsd';
    const title = 'Title';

    const wrapper = shallow(
        <GifGridItem 
            title={title} 
            url={url} 
        />
    );

    test('Test: Componente toMatchSnapshot()', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("Test: PÃ¡rafo Title", () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(title);
    });

    test("Test: img alt src", () => {
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);
    });

    test("Test: HTML content className", () => {
        const div = wrapper.find('div');
        const className = 'animate__tada';
        expect( div.prop('className').includes(className) ).toBe(true);
    });
});