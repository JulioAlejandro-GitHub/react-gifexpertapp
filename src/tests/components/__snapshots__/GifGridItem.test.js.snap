// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGridItem /> Test: Componente toMatchSnapshot() 1`] = `
<div
  className="card animate__animated animate__tada"
>
  <img
    alt="Title"
    src="httls://sdsd"
  />
  <p>
    Title
  </p>
</div>
`;
